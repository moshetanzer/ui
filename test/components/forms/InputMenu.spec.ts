import { describe, it, expect } from 'vitest'
import { UInputMenu } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('InputMenu', () => {
  it.each([
    ['basic case', {}],
    ['with name', { props: { name: 'username' } }],
    ['with placeholder', { props: { placeholder: 'Enter your username' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with required', { props: { required: true } }],
    ['with custom type', { props: { type: 'password' } }],
    ['with nullable', { props: { nullable: true } }],
    ['with id', { props: { id: 'comboboxId' } }],
    ['with leading', { props: { leading: true } }],
    ['with icon', { props: { icon: 'i-heroicons-battery-50-solid' } }],
    ['with leading and icon', { props: { leading: true, icon: 'i-heroicons-battery-50-solid' } }],
    ['with leadingIcon', { props: { leadingIcon: 'i-heroicons-battery-50-solid' } }],
    ['with loading icon', { props: { loading: true } }],
    ['with leading slots', { slots: { leading: () => 'leading slot' } }],
    ['with loading', { props: { loading: true } }],
    ['with trailing and icon', { props: { trailing: true, icon: 'i-heroicons-battery-50-solid' } }],
    ['with trailingIcon', { props: { trailing: true, trailingIcon: 'i-heroicons-battery-50-solid' } }],
    ['with loading and trailing', { props: { trailing: true, loading: true } }],
    ['with trailing slots', { slots: { leading: () => 'trailing slot' } }],
    ['with inputClass', { props: { inputClass: 'w-full' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UInputMenu.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UInputMenu)
    expect(html).toMatchSnapshot()
  })
})
