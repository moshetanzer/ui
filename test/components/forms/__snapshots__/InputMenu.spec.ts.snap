// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputMenu > renders basic case correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-0" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-1">
    <!--v-if--><button id="headlessui-combobox-button-1" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with custom type correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-10" aria-expanded="false" aria-autocomplete="list" role="combobox" type="password" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-11">
    <!--v-if--><button id="headlessui-combobox-button-11" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with disabled correctly 1`] = `
"<div class="relative" data-headlessui-state="disabled">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-6" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" disabled="" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-7">
    <!--v-if--><button id="headlessui-combobox-button-7" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" disabled="" data-headlessui-state="disabled" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with icon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-17" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-18"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-battery-50-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><button id="headlessui-combobox-button-18" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with id correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="comboboxId" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-14">
    <!--v-if--><button id="headlessui-combobox-button-14" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with inputClass correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-37" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9 w-full" autocomplete="off" aria-labelledby="headlessui-combobox-button-38">
    <!--v-if--><button id="headlessui-combobox-button-38" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with leading and icon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-19" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-20"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-battery-50-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><button id="headlessui-combobox-button-20" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with leading correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-15" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-16">
    <!--v-if--><button id="headlessui-combobox-button-16" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with leading slots correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-25" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-26"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5">leading slot</span><button id="headlessui-combobox-button-26" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with leadingIcon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-21" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-22"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-battery-50-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><button id="headlessui-combobox-button-22" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with loading and trailing correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-33" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-34">
    <!--v-if--><button id="headlessui-combobox-button-34" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-arrow-path-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with loading correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-27" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-28"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-arrow-path-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></span><button id="headlessui-combobox-button-28" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with loading icon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-23" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-24"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-arrow-path-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></span><button id="headlessui-combobox-button-24" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with name correctly 1`] = `
"<input style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;" type="hidden" hidden="" readonly="" name="username">
<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-2" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" name="username" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-3">
    <!--v-if--><button id="headlessui-combobox-button-3" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with nullable correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-12" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-13">
    <!--v-if--><button id="headlessui-combobox-button-13" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with placeholder correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-4" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" placeholder="Enter your username" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-5">
    <!--v-if--><button id="headlessui-combobox-button-5" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with required correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-8" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" required="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-9">
    <!--v-if--><button id="headlessui-combobox-button-9" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with trailing and icon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-29" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-30">
    <!--v-if--><button id="headlessui-combobox-button-30" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with trailing slots correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-35" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-36"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5">trailing slot</span><button id="headlessui-combobox-button-36" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with trailingIcon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <div class="inline-flex w-full"><input value="" id="headlessui-combobox-input-31" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-input rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-sm px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" autocomplete="off" aria-labelledby="headlessui-combobox-button-32">
    <!--v-if--><button id="headlessui-combobox-button-32" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-battery-50-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;
